src/handlers/VehicleTestController.ts

```diff
@@ -276,6 +276,13 @@ export class VehicleTestController implements IVehicleTestController {
     const newTestResult = cloneDeep(oldTestResult);
     oldTestResult.testVersion = enums.TEST_VERSION.ARCHIVED;
     newTestResult.testVersion = enums.TEST_VERSION.CURRENT;
+    if (payload.testTypes.length === 1) {
```

Comment:
I think this block of code can be simplified like this
```
if (payload.testTypes.length === 1 && testType[0].testNumber !== testNumber) {
      payload.testTypes = [...payload.testTypes, ...testTypes];
}

Reply:
Assuming this is what you mean:
```js
if (payload.testTypes.length === 1 && newTestResult.testTypes[0].testNumber !== payload.testTypes[0].testNumber) {
      payload.testTypes = [...newTestResult.testTypes, ...payload.testTypes];
}
```
It would be making an assumption about the shape of the data and that the testType being amended is not at index 0.

Reply:
yeah my mistake. Maybe rename the const variable name to something else rather than testTypes, maybe `newTestResultTestTypes`

Reply:
Done